src/nvim/tui/tui.c

```diff
@@ -48,10 +48,15 @@
 #define OUTBUF_SIZE 0xffff
 
 #define TOO_MANY_EVENTS 1000000
-#define STARTS_WITH(str, prefix) (strlen(str) >= (sizeof(prefix) - 1) \
-    && 0 == memcmp((str), (prefix), sizeof(prefix) - 1))
-#define TMUX_WRAP(is_tmux, seq) ((is_tmux) \
-    ? "\x1bPtmux;\x1b" seq "\x1b\\" : seq)
+#define STARTS_WITH(str, prefix) \
+  (strlen(str) >= (sizeof(prefix) - 1) && 0 == memcmp((str), (prefix), \
+                                                      sizeof(prefix) - 1))
+#define SCREEN_WRAP(is_screen, seq) ((is_screen) \
+                                     ? DCS_STR seq STREM_STR : seq)
+#define SCREEN_TMUX_WRAP(is_screen, is_tmux, seq) \
```

Comment:
would be nice to keep the TMUX_WRAP as well and express SCREEN_TMUX_WRAP via SCREEN_WRAP/TMUX_WRAP. I suppose the compiler can optimise redundant if, anyway it's just run on init.

Reply:
In the beginning I tried to implement it that way. But in that case, the expansion result is as follows.

```C
((screen) "\x1bP" ((tmux) ? "\x1bP "tmux;\x1b]50;\x07" "\x1b\\" : "\x1b]50;\x07") "\x1b\\"  :  ((tmux) ? "\x1bP" "tmux;\x1b]50;\x07" "\x1b\\" : "\x1b]50;\x07"));
```

This resulted in an error. I don't think this is allowed in C grammar when writing an expression in a string literal.